### PR TITLE
Test Conversion for Query Tag Tests

### PR DESCRIPTION
**Pull request by McKnight-42**
### Description

snowflake specific pytest conversion of integration test `query_tag_tests` to new format.

TODO:
- [x] Run python script
- [x] Reformat as needed
- [x] test locally
- [x] delete old test 

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: tests/functional/query_tag/test_query_tags.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            'snapshots': {
+                'tests': {
+                    'query_tag': self.prefix,
+                    'post-hook': '{{ check_tag() }}'
+                },
+            },
+        }
+
+    def build_all_with_query_tags(self, project):
+        run_dbt(['build', '--vars', '{{"check_tag": "{}"}}'.format(self.prefix)])
```
Comment (ChenyuLInx) [inline tests/functional/query_tag/test_query_tags.py:112]:
```suggestion
    @pytest.fixture(scope="class")
    def project_config_update(self, prefix):
        return {
            'config-version': 2,
            'models': {
                'tests': {
                    'query_tag': prefix,
                    'post-hook': '{{ check_tag() }}'
                },
            },
            'seeds': {
                'tests': {
                    'query_tag': prefix,
                    'post-hook': '{{ check_tag() }}'
                },
            },
            'snapshots': {
                'tests': {
                    'query_tag': prefix,
                    'post-hook': '{{ check_tag() }}'
                },
            },
        }

    def build_all_with_query_tags(self, project, prefix):
        run_dbt(['build', '--vars', '{{"check_tag": "{}"}}'.format(prefix)])
```

Comment (ChenyuLInx) [reply tests/functional/query_tag/test_query_tags.py:112]: requiring [prefix fixture](https://github.com/dbt-labs/dbt-core/blob/a7ff003d4f74763f126e77a0fc9c55e7a3311b3d/core/dbt/tests/fixtures/project.py#L62) should solve your problem

**Comment by McKnight-42** [comment]
tagging this pr to keep track on it https://github.com/dbt-labs/dbt-snowflake/pull/210 so we can update tests all at once.

**Review by ChenyuLInx — APPROVED** [review]
LGTM!

**Review by emmyoop — APPROVED** [review]

**Pull request merged**